cmd/devcli/commands/deployment/deployment.go

```diff
@@ -0,0 +1,321 @@
+		clio.Info(template)
+		ccfg, err := cfaws.ConfigFromContextOrDefault(ctx)
+		if err != nil {
+			return err
+		}
+		cfnClient := cfn.New(ccfg)
+
+		si := spinner.New(spinner.CharSets[14], 100*time.Millisecond)
+		si.Suffix = " creating CloudFormation change set"
+		si.Writer = os.Stderr
+		si.Start()
+
+		//todo update this from being hardcoded parameters for testvault
+		params := []aws_types.Parameter{
+			{
+				ParameterKey:   aws.String("UniqueVaultId"),
+				ParameterValue: aws.String("2CBsuomHFRE3mrpLGWFaxbyKXG6"),
+			},
+			{
+				ParameterKey:   aws.String("Id"),
+				ParameterValue: &deploymentName,
+			},
+			{
+				ParameterKey:   aws.String("BootstrapBucketName"),
+				ParameterValue: &bootstrapBucket,
+			},
+			{
+				ParameterKey:   aws.String("AssetPath"),
+				ParameterValue: aws.String(path.Join(providerPublisher, providerName, providerVersion, "handler.zip")),
+			},
+			{
+				ParameterKey:   aws.String("ApiUrl"),
+				ParameterValue: aws.String("https://prod.testvault.granted.run"),
+			},
+			{
+				ParameterKey:   aws.String("AccountId"),
+				ParameterValue: &accountId,
+			},
+		}
+
+		stackName := providerName + suffix
+
+		changeSetName, createErr := cfnClient.CreateChangeSet(ctx, template, params, nil, stackName, "")
+
+		si.Stop()
+
+		if createErr != nil {
+
+			return createErr
+
+		}
+
+		uiClient := ui.New(ccfg)
+
+		status, err := uiClient.FormatChangeSet(ctx, stackName, changeSetName)
+		if err != nil {
+			return err
+		}
+		clio.Info("The following CloudFormation changes will be made:")
+		fmt.Println(status)
+
+		err = cfnClient.ExecuteChangeSet(ctx, stackName, changeSetName)
+		if err != nil {
+			return err
+		}
+
+		status, messages := uiClient.WaitForStackToSettle(ctx, stackName)
+
+		fmt.Println("Final stack status:", ui.ColouriseStatus(status))
+
+		if len(messages) > 0 {
+			fmt.Println(console.Yellow("Messages:"))
+			for _, message := range messages {
+				fmt.Printf("  - %s\n", message)
+			}
+		}
+
+		//create target group
+
+		cfApi, err := types.NewClientWithResponses(cfg.CommonFateAPIURL)
+		if err != nil {
+			return err
+		}
+
+		targetGroupId := c.String("target-group-override")
+
+		//if there was an override dont create a new target group but link with an old one
+		if targetGroupId == "" {
+			tgCreateReq := types.CreateTargetGroupJSONRequestBody{
+				ID: providerName + "-" + providerVersion + "-" + suffix,
+				// will create the target group as eg. commonfate-v0.1.4-jacktest
+				TargetSchema: assetString,
+			}
+			_, err = cfApi.CreateTargetGroupWithResponse(ctx, tgCreateReq)
+			if err != nil {
+				return err
+			}
+			clio.Successf("Successfully created target group '%s'", tgCreateReq.ID)
+			targetGroupId = tgCreateReq.ID
+
+		}
+
+		reqBody := types.CreateTargetGroupDeploymentJSONRequestBody{
+			AwsAccount: accountId,
+			AwsRegion:  awsRegion,
+			Id:         deploymentName,
+			Runtime:    "aws",
```

Comment:
not flagging this as relevant to your PR, but we might want make this an enum for type safety later on

Reply:
This should really be an input, will update that now